src/components/Header/Header.tsx

```diff
@@ -0,0 +1,90 @@
+import { FC, Dispatch, SetStateAction, useRef, useEffect } from 'react';
+import cn from 'classnames';
+import { InputForm } from '../InputForm';
+import { Todo } from '../../types/Todo';
+import { Errors } from '../../types/Errors';
+import { handleError } from '../../utils/handleError';
+import { postTodo, USER_ID } from '../../api/todos';
+
+type Props = {
+  uncompletedTodosAmount: number;
+  todos: Todo[];
+  tempTodo: Todo | null;
+  setTempTodo: Dispatch<SetStateAction<Todo | null>>;
+  setError: Dispatch<SetStateAction<Errors>>;
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+};
+
+export const Header: FC<Props> = ({
+  setTempTodo,
+  uncompletedTodosAmount,
+  tempTodo,
+  todos,
+  setError,
+  setTodos,
+}) => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, tempTodo]);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const inputValue = inputRef.current?.value.trim();
+
+    if (!inputValue) {
+      handleError(setError, Errors.EmptyTitle);
+
+      return;
+    }
+
+    const temporaryTodo: Todo = {
+      id: 0,
+      userId: USER_ID,
+      title: inputValue,
+      completed: false,
+    };
+
+    setTempTodo(temporaryTodo);
+
+    postTodo(temporaryTodo)
+      .then(res => {
+        setTodos(current => [...current, res]);
+
+        if (inputRef.current?.value) {
+          inputRef.current.value = '';
+        }
```

Comment:
After posting the `todo`, you clear the input by setting` inputRef.current.value` to an empty string. This is effective but consider using `setState `for controlled components if you're maintaining form state with React. 
It provides better control and may prevent issues when the component re-renders.